index.html

```diff
@@ -0,0 +1,475 @@
+
+CSDL introduces a suite of types and directives. All CSDL-specific types and directives are prefixed with `csdl_` as recommended in [the GraphQL specification](https://spec.graphql.org/draft/#note-ca863).
+
+### Using this spec
+
+CSDL documents **must** contain a `schema` declaration with a [`@using` directive](/using/draft/pre-0).
+
+```graphql
+#::[`@using` directive declaring that the document conforms to csdl v1]
+schema @using(spec: "https://specs.apollo.dev/csdl/v1")
+```
+
+### Forbidden GraphQL IDL features
+
+CSDL documents cannot contain certain [GraphQL type system](https://spec.graphql.org/draft/#sec-Type-System) constructs. Specifically, **extensions are forbidden.** You cannot `extend type` or `extend interface` within a CSDL document. Producers **must** produce documents without extensions.
+
+## Data Model
+TK, draws heavily from federation, a little more general.
+
+### Realized types
+
+A type is *realized* wherever a resolver from a particular *subgraph* returns that type. We use the notation `subgraph::Type` to represent `Type` realized by `subgraph` and `subgraph::Type.field` to represent the resolver for `Type.field` within `subgraph`.
+
+```graphql
+#::[Query with resolvers and realized types]
+query {          # resolver                    | realizes
+                 # ----------------------------|--------------
+  rockets {      # rockets::Query.rockets     -> rockets::Rocket
+    captain {    # rockets::Rocket.astronaut  -> rockets::Astronaut
+      name       # astronauts::Astronaut.name -> String
+      rocket {   # rockets::Astronaut.rocket  -> rockets::Rocket
+        id       # rockets::Rocket.id         -> ID
+      }
+    }
+  }
+
+}
+```
+
+These distinctions are immaterial to the client. As far as clients are concerned, all selections on `Rocket` are the same.
+
+This illusion is maintained by the router, which must break incoming queries into subqueries. In the example above, the path `rockets.captain` is resolved by the `rockets::Rocket.astronaut` resolver, so its selection set is realized as a `rockets::Astronaut`. However, there is no `name` field for `rockets::Astronaut`—instead, the only resolver for `Astronaut.name` is `astronauts::Astronaut.name`. To process this query, the router must "convert" a `rockets::Astronaut` to an `astronauts::Astronaut` via an `_entities` query.
+
+### Portability
+A type is *portable to* a subgraph if it has a `@csdl_key` for that subgraph.
+
+### Free / Bound fields
+
+Free fields can be resolved by any subgraph, bound fields can only be resolved by the subgraph they're bound to.
+
+### Mandatory Types
+
+A CSDL document **must** define a `csdl_Graph` enum. Each enum value defines a subgraph. Each value **must** be annotated with a `@csdl_endpoint` directive specifying the endpoint URL for the subgraph.
```

Comment:
I really like the enum idea.

Should the name be csdl_Subgraph though? On the one hand `csdl_Graph` makes sense in that the entire enum declaration is kinda declaring the graph. On the other hand, usually the name of an enum is the category that the enum values fall into.